posthog/poller.py

```diff
@@ -4,7 +4,7 @@
 class Poller(threading.Thread):
     def __init__(self, interval, execute, *args, **kwargs):
         threading.Thread.__init__(self)
-        self.daemon = False
+        self.daemon = True  # Make daemon to not interfere with program exit
```

Comment:
Do we need to enable this given we're now stopping the poller on exit?

Reply:
Indeed.

It's only when we manually shut down the runner, that the onExit part is called.

In other cases, given nothing is actively running, we don't want poll to actively block exit (which is what it does when it's not a daemon thread).
The program automatically exits when the only things running are daemon threads.